aiidalab_widgets_base/computational_resources.py

```diff
@@ -425,8 +425,15 @@ def _is_in_config(self):
         fpath = Path.home() / ".ssh" / "config"
         if not fpath.exists():
             return False
-        cfglines = open(fpath).read().split("\n")
-        return "Host " + self.hostname.value in cfglines
+        sshcfg = parse_sshconfig(self.hostname.value)
+        # Confusingly, parse_sshconfig returns a dict with a hostname
+        # even if it is not in the config file.
```

Comment:
I am not sure why it is confusing. This is what SSH does itself. If the `Hostname` is not specified, the value is taken directly from the `Host`

Reply:
I am not sure what you mean by `Host`. The function is called `parse_sshconfig` and we pass it the hostname that we're interested in. I would expect that if the given hostname is not in the config file, that this function returns None or something like that. Anyway, I've reworded the comment a bit , but I would like to keep it to make things clearer.

Reply:
> I am not sure what you mean by `Host`.

I was referring to the ssh configuration. Suppose it looks like that.
```
Host my_computer
    HostName my_computer.com
    User me
```

In that context, if the `HostName` (`my_computer.com`) is not set in the `~/.ssh/config`, it is taken directly from the `Host` (`my_computer`). Consequently, if nothing is provided at all, the ssh command `ssh my_computer` (where `my_computer` is the `Host`) would imply that you are requesting a `Hostname` `my_computer`.

So, it is not surprising to me that if you request an SSH configuration for a host "whatever", the `Hostname` setting can be identified already based on your request.

That is how I would interpret the behaviour of the function.

Reply:
Interesting, I was unaware of the difference between Host and Hostname. Thanks!